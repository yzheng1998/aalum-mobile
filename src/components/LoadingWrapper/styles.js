import styled from 'styled-components'

export const Container = styled.View`
  flex: 1;
`

export const CenteredContainer = styled.View`
  flex: 1;
  padding: 30px;
  align-items: center;
  justify-content: center;
`
