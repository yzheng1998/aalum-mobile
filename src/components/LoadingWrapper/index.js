import React, { Component } from 'react'
import { ActivityIndicator } from 'react-native'
import { Container, CenteredContainer } from './styles'

export default class LoadingWrapper extends Component {
  render() {
    const { children, loading, style } = this.props
    return (
      <Container>
        {loading ? (
          <CenteredContainer>
            <ActivityIndicator />
          </CenteredContainer>
        ) : (
          <Container style={style}>{children}</Container>
        )}
      </Container>
    )
  }
}
