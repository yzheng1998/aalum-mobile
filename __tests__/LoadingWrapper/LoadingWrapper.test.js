import React from 'react'
import LoadingWrapper from '../../src/components/LoadingWrapper'
import styled from 'styled-components'
import { shallow } from 'enzyme'

describe('LoadingWrapper', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<LoadingWrapper />)
    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({ loading: true })
    expect(wrapper).toMatchSnapshot()
  })

  test('renders children properly', () => {
    const MockChild1 = styled.View``
    const MockChild2 = styled.View``
    MockChild1.displayName = 'MockChild1'
    MockChild2.displayName = 'MockChild2'
    const wrapper = shallow(
      <LoadingWrapper loading>
        <MockChild1 />
        <MockChild2 />
      </LoadingWrapper>
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('MockChild1').length).toBe(0)
    expect(wrapper.find('MockChild2').length).toBe(0)

    wrapper.setProps({ loading: false })
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('MockChild1').length).toBe(1)
    expect(wrapper.find('MockChild2').length).toBe(1)
  })

  test('renders style properly', () => {
    const styleMock = { color: '#ffffff', backgroundColor: '#000000' }
    const wrapper = shallow(
      <LoadingWrapper loading={false} style={styleMock} />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
