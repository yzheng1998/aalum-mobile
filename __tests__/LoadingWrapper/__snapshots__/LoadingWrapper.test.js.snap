// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingWrapper renders children properly 1`] = `
<Styled(View)>
  <Styled(View)>
    <Component
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
  </Styled(View)>
</Styled(View)>
`;

exports[`LoadingWrapper renders children properly 2`] = `
<Styled(View)>
  <Styled(View)>
    <MockChild1 />
    <MockChild2 />
  </Styled(View)>
</Styled(View)>
`;

exports[`LoadingWrapper renders correctly 1`] = `
<Styled(View)>
  <Styled(View) />
</Styled(View)>
`;

exports[`LoadingWrapper renders correctly 2`] = `
<Styled(View)>
  <Styled(View)>
    <Component
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
  </Styled(View)>
</Styled(View)>
`;

exports[`LoadingWrapper renders style properly 1`] = `
<Styled(View)>
  <Styled(View)
    style={
      Object {
        "backgroundColor": "#000000",
        "color": "#ffffff",
      }
    }
  />
</Styled(View)>
`;
